build.assets/tooling/lib/eventschema/schemagen.go

```diff
@@ -0,0 +1,256 @@
+/*
+Copyright 2021-2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package eventschema
+
+import (
+	"strings"
+
+	"github.com/gravitational/trace"
+	apiextv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+
+	"github.com/gravitational/teleport/build.assets/tooling/lib/tree"
+)
+
+// TODO
+// index by event type (static mapping for now)
+// how can I know which event has a view? -> maintain a list of ignored events -> check ignore logic
+// AST or protobuf?
+// get the schema from API ?
+// hide metadata
+// use hardcoded event_date event_time and other fields
```

Comment:
What is the plan for this? Are you going to address those before or after this PR is merged?

Reply:
This is a leftover of my personal notes when this was a WIP. All those steps have been addressed except the "hide metadata" part which will be part of a subsequent "event field masking" PR.